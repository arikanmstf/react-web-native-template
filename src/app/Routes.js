// @flow
import React from 'react';
import { HOME, DUMMY_TWO } from 'constants/routes/routeNames';
import type { RoutesProps } from './RoutesTypes';

const Routes = (props: RoutesProps) => {
  const {
    DummyScreen, DummyScreenTwo, Router, Switch, Route,
  } = props;

  return (
    <Router>
      <Switch>
        <Route exact path={HOME} component={DummyScreen} />
        <Route exact path={DUMMY_TWO} component={DummyScreenTwo} />
      </Switch>
    </Router>
  );
};

export default Routes;
