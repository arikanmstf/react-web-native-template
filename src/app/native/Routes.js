import React from 'react';
import { MemoryRouter as Router, Switch, Route } from 'react-router-native';
import { HOME, DUMMY_TWO } from 'constants/routes/routeNames';

// Screens
import DummyScreen from 'screens/dummy/DummyNativeContainer';
import DummyScreenTwo from 'screens/dummyTwo/DummyNativeContainer';

const Routes = () => (
  <Router>
    <Switch>
      <Route exact path={HOME} component={DummyScreen} />
      <Route exact path={DUMMY_TWO} component={DummyScreenTwo} />
    </Switch>
  </Router>
);

export default Routes;
