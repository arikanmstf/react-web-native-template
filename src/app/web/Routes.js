import React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { HOME, DUMMY_TWO } from 'constants/routes/routeNames';

// Screens
import DummyScreen from 'screens/dummy/DummyWebContainer';
import DummyScreenTwo from 'screens/dummyTwo/DummyWebContainer';

const Routes = () => (
  <Router>
    <Switch>
      <Route exact path={HOME} component={DummyScreen} />
      <Route exact path={DUMMY_TWO} component={DummyScreenTwo} />
    </Switch>
  </Router>
);

export default Routes;
