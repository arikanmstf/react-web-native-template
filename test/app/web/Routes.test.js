import React from 'react';
import { shallow } from 'enzyme';
import Routes from 'web/Routes';

describe('src/app/web/Routes.js', () => {
  it('Render: props: null', () => {
    const wrapper = shallow(<Routes />);
    expect(wrapper).toMatchSnapshot();
  });
});
