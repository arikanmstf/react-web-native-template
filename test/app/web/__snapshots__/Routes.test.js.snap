// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/app/web/Routes.js Render: props: null 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Routes />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/dummy-two"
        />
      </Switch>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/dummy-two"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/dummy-two",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/dummy-two"
          />
        </Switch>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/dummy-two"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/dummy-two",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
